src/client/players.py

```diff
@@ -45,19 +45,25 @@ def isFriend(self, id):
         '''
         Convenience function for other modules to inquire about a user's friendliness.
         '''
+        if id in self._logins:
+            id = self._logins[id].id
```

Comment:
It's almost done, and I do believe friends are queried entirely by id.

The current `players['Name']` implementation does do fallback lookup by strings, albeit not in the most efficient manner. It should be fine for now though.